### PR TITLE
[EMBR-4352] Run Swiftlint/Clang-format over RN iOS module

### PR DESCRIPTION
**Pull request by facostaembrace**
## Test

> given this [failure](https://github.com/embrace-io/embrace-react-native-sdk/actions/runs/10534470261/job/29192402283) had to revert this format [change](https://github.com/embrace-io/embrace-react-native-sdk/pull/83/commits/5b7922171b74ddbbd0a135f220ddf48fd023d683). To analyze how this string is working and if it is safe just to fix the test with the new format of the file.

UPDATE: previous issue was fixed. Unit tests work fine after tweaks on strings.

## Goal

- adding Swiftlint for swift files (`yarn lint:swift`)
- adding linter for Objective-c files (`yarn lint:clang`)
- adding script for linting js files only into packages/** directory (`yarn lint:js`). NOTE: the base `yarn lint` cover json files, and json files are also into ios/android directories making the plugin to run into the entire project (resulting in unexpected behaviors)
- adding husky to run linters when commiting

### CODE REVIEW

File: package.json
Comment (jpmunz) [inline package.json:17]: is this regular `lint` command still needed given the others we have now?

File: packages/core/ios/RNEmbrace/EmbraceManager.m
Comment (jpmunz) [inline packages/core/ios/RNEmbrace/EmbraceManager.m:1]: this one will conflict since we fully changed it on 5.0, might be better to not update as part of this PR

File: packages/core/scripts/util/ios.ts
Comment (jpmunz) [inline packages/core/scripts/util/ios.ts:22]: I'm not sure about this change, since it's injected into the customer's app delegate I think this will end up including extra whitespace

**Review by jpmunz — APPROVED** [review]
lgtm, I think this would be good to hook up to the github workflow as well

when merging 5.0 we should remember remove the extra `.swiftlint.yml` files added there

File: packages/core/ios/RNEmbrace/EmbraceManager.m
Comment (facostaembrace) [reply packages/core/ios/RNEmbrace/EmbraceManager.m:1]: Yes, this is why I would prefer to wait until we make the releases and then after all merge this pr

File: package.json
Comment (jpmunz) [reply package.json:17]: ya I think we can just remove

**Pull request closed without merging**
